src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -74,12 +80,19 @@ private static ExchangeAlgorithmType MapExchangeAlgorithmType(long mac)
             {
                 case 1:
                     return ExchangeAlgorithmType.RsaKeyX;
-                case 2 | 32:
-                    return ExchangeAlgorithmType.RsaSign;
-                case 128:
+                case 2:
+                case 4:
+                case 8:
                     return ExchangeAlgorithmType.DiffieHellman;
+                case 16:
+                case 32:
+                case 64:
+                case 128:
+                case 256:
+                case 512:
+                case 1024:
+                    return (ExchangeAlgorithmType) mac;
```

Comment:
Same as MapCipherAlgorithmType.  We need to not have two different unmapped values for the same thing.

For both of them, if OpenSSL exposes a concept Windows doesn't, we should find a way to update the enum (it's a mostly-non-breaking change, but the public names need to go through API review).  The only breaking-ness is that someone referencing CoreFX wouldn't be able to go back to Desktop with that identifier.